x-pack/functionbeat/provider/aws/aws/transformer/transformer.go

```diff
@@ -210,3 +218,82 @@ func SQS(request events.SQSEvent) []beat.Event {
 	}
 	return events
 }
+
+func S3GetEvents(request events.S3Event) ([]beat.Event, error) {
+	var evts []beat.Event
+	svc := s3.New(session.New())
+	for _, record := range request.Records {
+		unescaped_key, err := url.QueryUnescape(record.S3.Object.Key)
+
+		if err != nil {
+			fmt.Println("Got error unescaping key: %s", record.S3.Object.Key)
+			fmt.Println(err.Error())
+			return nil, err
+		}
+
+		result, err := svc.GetObject(&s3.GetObjectInput{
+			Bucket: aws.String(record.S3.Bucket.Name),
+			Key:    aws.String(unescaped_key),
+		})
+
+		if err != nil {
+			fmt.Println("Got error calling GetObject:")
+			fmt.Println(err.Error())
+			return nil, err
+		}
+
+		defer result.Body.Close()
+		obj, err := ioutil.ReadAll(result.Body)
+
+		if err != nil {
+			fmt.Println(err)
+			return nil, err
+		}
+
+		if obj[0] == 31 && obj[1] == 139 {
+			inBuf := bytes.NewBuffer(obj)
+			r, err := gzip.NewReader(inBuf)
+			if err != nil {
+				return nil, err
+			}
+
+			var outBuf bytes.Buffer
+			_, err = io.Copy(&outBuf, r)
+			if err != nil {
+				r.Close()
+				return nil, err
+			}
+
+			err = r.Close()
+			if err != nil {
+				return nil, err
+			}
+			obj = outBuf.Bytes()
+		}
+
+		obj_r := bytes.NewReader(obj)
```

Comment:
Please do not use snake case variable names because it is not idiomatic Golang: https://golang.org/doc/effective_go#mixed-caps

Reply:
done ;)